app/api/orders.py

```diff
@@ -319,7 +317,7 @@ def before_update_object(self, order, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-        if data.get('amount') or data.get('is_billing_enabled'):
+        if data.get('amount') and (data.get('is_billing_enabled') or order.event.is_billing_info_mandatory):
```

Comment:
This will again, require billing info even if the organizer has asked not to. Please read the issue again. I don't think you understand what the goal is.

Can you tell me what is the goal of this PR?